src/mlpack/methods/reinforcement_learning/replay/episodic_replay.hpp

```diff
@@ -0,0 +1,249 @@
+#define MLPACK_METHODS_RL_REPLAY_EPISODIC_REPLAY_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of episodic experience replay.
+ *
+ * Each episode of interactions between the agent and the
+ * environment will be saved to a memory buffer. When necessary,
+ * we can simply sample previous episodes or the most recent episode
+ * from the buffer to train the agent.
+ *
+ * @tparam EnvironmentType Desired task.
+ */
+template <typename EnvironmentType>
+class EpisodicReplay
+{
+ public:
+  //! Convenient typedef for action.
+  using ActionType = typename EnvironmentType::Action;
+
+  //! Convenient typedef for state.
+  using StateType = typename EnvironmentType::State;
+
+  /**
+  * Construct an instance of episode experience replay class.
+  */
+  EpisodicReplay():
+      capacity(0),
+      position(0),
+      max_episode_len(0),
+      full(false)
+  { /* Nothing to do here. */ }
+
+  /**
+  * Construct an instance of episode experience replay class.
+  *
+  * @param capacity Total memory size in terms of number of episodes.
+  * @param max_episode_len The maximum episode length possible.
+  * @param dimension The dimension of an encoded state.
+  */
+  EpisodicReplay(const size_t capacity,
+                 const size_t max_episode_len,
+                 const size_t dimension = StateType::dimension) :
+      capacity(capacity),
+      position(0),
+      max_episode_len(max_episode_len),
+      full(false)
+  {
+    states.resize(capacity);
+    next_states.resize(capacity);
+    rewards.resize(capacity);
+    actions.resize(capacity);
+    isTerminal.resize(capacity); 
+  }
+
+
+  /**
+  * Store the given experience.
+  *
+  * @param state Given state.
+  * @param action Given action.
+  * @param reward Given reward.
+  * @param nextState Given next state.
+  * @param isEnd Whether next state is terminal state.
+  */
+  void Store(const StateType& state,
+             ActionType action,
+             double reward,
+             const StateType& nextState,
+             bool isEnd)
+  {
+    if (clear)
+    {
+      states[position].clear();
+      actions[position].clear();
+      rewards[position].clear();
+      next_states[position].clear();
+      isTerminal[position].clear();
+      clear = false;
+    }
+
+    states[position].push_back(state.Encode());
+    actions[position].push_back(action);
+    rewards[position].push_back(reward);
+    next_states[position].push_back(nextState.Encode());
+    isTerminal[position].push_back(isEnd);
+
+    if(isEnd || states[position].size()==max_episode_len)
+    {
+      position++;
+      clear = true;
+    }
+    if (position == capacity)
+    {
+      full = true;
+      position = 0;
+    }
+  }
+
+  /**
+  * Get the number of episodes in the memory.
+  *
+  * @return Actual used memory size
+  */
+
+  const size_t Size()
+  {
+    if(states[position].size()==0){
```

Comment:
Could you start the bracket from a new line, and give appropriate spaces as this.
`if (states[position].size() == 0)`
Thanks.